### PR TITLE
fix: fixes bug for first table construction

### PR DESCRIPTION
**Pull request by Specy**

### CODE REVIEW

File: assets/grammars/correct/indirect_empty.lr [hunk 1]
```diff
@@ -0,0 +1,5 @@
+S -> A B
+A -> C
+C -> D
+D -> ''
+B -> 'x'
```
Comment (umut-sahin) [inline assets/grammars/correct/indirect_empty.lr:5]: This file needs a trailing newline.

Comment (umut-sahin) [reply assets/grammars/correct/indirect_empty.lr:5]: Also, could you put some blank lines between rule groups:
```
S -> A B

A -> C
C -> D
D -> ''

B -> 'x'
```

File: src/tables.rs [hunk 2]
```diff
@@ -12,7 +12,7 @@ impl FirstTable {
     /// Constructs the first table from the grammar.
     pub fn construct(grammar: &Grammar) -> FirstTable {
         let mut first_table = IndexMap::new();
-
+        let mut indirectly_empty_symbols = IndexSet::new();
         let mut done = false;
```
Comment (umut-sahin) [inline src/tables.rs:16]: Blank line between these two as well.

File: tests/parser.rs [hunk 3]
```diff
@@ -144,6 +144,96 @@ fn raising_correct_error_when_creating_lalr_parser_for_non_lalr_grammar() {
 }
 
 
+#[test]
+#[cfg_attr(target_family = "wasm", wasm_bindgen_test)]
+fn correctly_creates_first_and_follow_sets_for_indirectly_empty_grammar() {
```
Comment (umut-sahin) [inline tests/parser.rs:149]: `correctly_creating_first_and_follow_sets_for_indirectly_empty_grammar`

File: tests/parser.rs [hunk 4]
```diff
@@ -144,6 +144,96 @@ fn raising_correct_error_when_creating_lalr_parser_for_non_lalr_grammar() {
...
+    let grammar = Grammar::parse(common::grammars::INDIRECT_EMPTY).unwrap();
+    let parser = Parser::lr(grammar).unwrap();
+    let first_table = parser.first_table();
```
Comment (umut-sahin) [inline tests/parser.rs:152]: Blank line between these two.

**Review by umut-sahin — CHANGES_REQUESTED** [review]
Should we also have some tests for correctly tracing/parsing with indirectly empty symbols? Just in case something else is buggy.

**Review by umut-sahin — APPROVED** [review]
